impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -354,11 +392,39 @@ public synchronized void close() throws Exception {
         }
     }
 
+    /**
+     * Workaround for <a href="https://issues.apache.org/jira/browse/KAFKA-14287">KAFKA-14287</a>.
+     * with kraft, if we don't shut down the controller last, we sometimes see a hang.
+     */
+    private void roleOrderedShutdown(Map<Integer, Server> servers, boolean await) {
+        var justBrokers = servers.entrySet().stream()
+                .filter(e -> !portsAllocator.hasRegisteredPort(Listener.CONTROLLER, e.getKey()))
+                .map(Map.Entry::getValue)
+                .toList();
+        justBrokers.forEach(Server::shutdown);
+        if (await) {
+            justBrokers.forEach(Server::awaitShutdown);
+        }
```

Comment:
I can't decide if we should do `stoppedServers.putAll(justBrokers)` here or if doing it after this method returns is fine... 

I think its moot.